network/identity.go

```diff
@@ -144,13 +181,20 @@ func (i *identity) handleConnected(peerID peer.ID) error {
 		return ErrInvalidChainID
 	}
 
-	i.srv.addPeer(peerID)
+	if !resp.TemporaryDial && !status.TemporaryDial {
```

Comment:
This adds a peer only if both nodes are doing a temporary dial.

Peers won't be added if one node is doing a temporary dial, and the other is doing a persistent dial

Is this the behavior we want?

cc @Kourin1996

Reply:
@zivkovicmilos We only want to add a peer if both the participants wants to make a persistent dial because the sole purpose of this logic is to make  temporary call and find peers. BTW  If Node-1 is initiating  a temporary dial, Node-2 can't make a persistent dial, because node-1 marks this peer as a temporary dial peer.